jsonschema2pojo-maven-plugin/src/main/java/org/jsonschema2pojo/maven/Jsonschema2PojoMojo.java

```diff
@@ -1114,6 +1117,31 @@ public boolean isIncludeAdditionalProperties() {
     @Override
     public boolean isIncludeSetters() { return includeSetters; }
 
+    private void setTargetVersion() {
+        if (isNotBlank(this.targetVersion)) {
+            return;
+        }
+
+        if (project.getProperties() != null && project.getProperties().containsKey("maven.compiler.target")) {
```

Comment:
Wouldn't that be provided by `defaultValue = "${maven.compiler.target}"` ?

Reply:
I thought best to centralise this logic in one place, so we can easily see what the order of precedence is and we are not relying on Maven machinery.

Reply:
I had originally implemented this inside getTargetVersion, but since this becomes quire a complex process I think best to trigger it once. I realise now that I replaced `return` with `this.targetVersion =` so I need to adjust this to ensure this process stops when it should.